sov-esdt-safe/src/lib.rs

```diff
@@ -18,8 +18,25 @@ pub trait SovEsdtSafe:
     + multiversx_sc_modules::pause::PauseModule
 {
     #[init]
-    fn init(&self, config: EsdtSafeConfig<Self::Api>) {
-        self.esdt_safe_config().set(config);
+    fn init(
+        &self,
+        fee_market_address: ManagedAddress,
+        opt_config: OptionalValue<EsdtSafeConfig<Self::Api>>,
+    ) {
+        self.require_sc_address(&fee_market_address);
+        self.fee_market_address().set(fee_market_address);
+
+        self.esdt_safe_config().set(
+            opt_config
+                .into_option()
+                .unwrap_or_else(EsdtSafeConfig::default_config),
+        );
+    }
+
+    #[only_owner]
+    #[endpoint(updateConfiguration)]
+    fn update_configuration(&self, new_config: EsdtSafeConfig<Self::Api>) {
+        self.esdt_safe_config().set(new_config);
```

Comment:
Same check that `new_config` is valid. Maybe we should have a common module for this if possible